File: src/main/kotlin/net/merayen/elastic/backend/logicnodes/list/signalgenerator_1/LogicNode.java
package net.merayen.elastic.backend.logicnodes.list.signalgenerator_1;

import net.merayen.elastic.backend.logicnodes.Format;
import net.merayen.elastic.backend.nodes.BaseLogicNode;
import net.merayen.elastic.system.intercom.InputFrameData;
import net.merayen.elastic.system.intercom.NodeDataMessage;
import net.merayen.elastic.system.intercom.OutputFrameData;

public class LogicNode extends BaseLogicNode {
	@Override
	protected void onCreate() {
		createPort(new BaseLogicNode.PortDefinition() {{
			name = "frequency";
		}});

		createPort(new BaseLogicNode.PortDefinition() {{
			name = "output";
			format = Format.AUDIO;
			output = true;
		}});

		set("data.frequency", 440.0f);
	}

	@Override
	protected void onInit() {}

	@Override
	protected void onParameterChange(String key, Object value) { // Parameter change from UI
		set(key, value); // Acknowledge anyway
	}

	@Override
	protected void onConnect(String port) {}

	@Override
	protected void onDisconnect(String port) {}

	@Override
	protected void onFinishFrame(OutputFrameData data) {}

	@Override
	protected void onRemove() {}

	@Override
	protected void onData(NodeDataMessage data) {}
}

File: src/main/kotlin/net/merayen/elastic/backend/logicnodes/list/signalgenerator_1/LogicNode.kt
package net.merayen.elastic.backend.logicnodes.list.signalgenerator_1

import net.merayen.elastic.backend.logicnodes.Format
import net.merayen.elastic.backend.nodes.BaseLogicNode
import net.merayen.elastic.backend.nodes.BaseNodeData
import net.merayen.elastic.system.intercom.NodeDataMessage
import net.merayen.elastic.system.intercom.OutputFrameData

class LogicNode : BaseLogicNode() {
	override fun onCreate() {
		createInputPort("frequency")
		createOutputPort("output", Format.AUDIO)

		(properties as Data).frequency = 440f
	}

	override fun onInit() {}

	override fun onParameterChange(instance: BaseNodeData) { // Parameter change from UI
		this.updateProperties(instance) // Accept anyway
	}

	override fun onConnect(port: String) {}
	override fun onDisconnect(port: String) {}
	override fun onFinishFrame(data: OutputFrameData?) {}
	override fun onRemove() {}
	override fun onData(data: NodeDataMessage) {}
}
